### ISSUE TITLE
[BUG] Mouse buttons 3, 4, 5 etc. No longer recognized by BTT

### ISSUE THREAD

**Issue by philosoralphter** [fifafu/BetterTouchTool#862]
**IMPORTANT:** Don't post private requests here (e.g. license issues, etc.). Please send me an email at boastr.net@gmail.com instead.

---

#### Description of bug/feature request/question:
Cannot get btt to register clicks from buttons 3,4,5 in the clickbox in setup tab for "other mice". Worked fine before today.

Upon booting at work today, the extra mouse buttons I usually use for move a space left and move a space right are no longer registered in BTT.  I believe this was not happening when I updated to os 10.12.4, but rather a few days after that.   Also, looks like middle mouse button is not differentiated from left click either, but is recognized.

#### Affected input device (e.g. MacBook Trackpad, Magic Mouse/Trackpad, Touch Bar, etc.):
Evoluent ergonomic vertical mouse.  Weird, I know, but it used to work! And the buttons are still registered in OS (they do random things like move selection between icons on desktop.)

#### Device information:

Type of Mac: MBP retina 2GHz i7 
macOS version: 10.12.4 (updated on 25th, problem appeared may 1.
BetterTouchTool version: 2.095, 2.094, 2.071, 1.963 (tested all version left in my trash)

#### Additional information (e.g. StackTraces, related issues, screenshots, workarounds, etc.):

---

**Note:** Before bug reporting, please make sure you have the latest version of BetterTouchTool and that you have already tried to restart your system :-). If you encounter a crash, please attach a crash log from the macOS Console.app from the "User Diagnostic Reports" section.

**Comment by fifafu** [comment]
I don't think this is a bug in BTT.
Has anything changed with your mouse driver? Have you tried to restart your machine again?
What kind of mouse are you using?

**Comment by philosoralphter** [comment]
Wow thanks for the response!

It's an evoluent ergonomic mouse.  https://evoluent.com/products/vm4rw/  I didn't install any special drivers from their site, I let the OS handle it which it did.  It's a standard dongle mouse (skipped bluetooth and "mac" options) plug-n-play worked.  Maybe i'll go find their driver for it.

I agree it could be a change in some other code, but the computer was asleep since friday, and the last OS updates it got were last Wednesday after which it still worked fine until a restart. 

maybe the restart actually applied some other update that had been waiting.  any chance 10.12.4 has altered anything for you?  It's weird, again the buttons work in other contexts, but btt doesn't recognize them in the little test box at all...new accessibility permissions or something?

Thanks!!

**Comment by fifafu** [comment]
I don't think there has been a change in macOS that could cause this (my mice are still being recognized correctly).

Could you check whether BTT is still enabled in System Preferences => Security & Privacy => Privacy => Accessibility?

**Comment by philosoralphter** [comment]
Yeah it still is enabled/authorized for accessibility there.  I'm going to do another restart, look around for some mouse drivers, and get back to you.

**Comment by philosoralphter** [comment]
Well nothing yet.  here's a similar thing on apple discussions: https://discussions.apple.com/thread/7905735?start=0&tstart=0

and oh hey, looks like one of the users is using my mouse and has a solution including installing a  new driver they have.

I think there was something messed with in the kernel usb stuff in 10.12.14, there are all kind of reports like this with not much info or help just, "it's broken".

Thanks again for your help i'll let you know what I find, and then i'll close this ticket if you like (or we can leave it for others with same problem)

**Comment by fifafu** [comment]
That's interesting, so maybe 10.12.4 really broke something for some mice. If you find anything, please let me know here.

Apple today released a public beta of 10.12.5, but I haven't checked whether they mention any mouse related fixes.

**Comment by philosoralphter** [comment]
ok i followed the last guy's suggestions, and it did not resolve with the latest driver from the manufacturer.  Not sure I am having the same problem as the button is recognized (moves selections, etc), just not in btt.

How can i make sure i clear all settings from btt and reinstall? I think i may have a lot of plist baggage or whatever especially from trying with all the versions today.  I can't get it to turn off window snapping, and I just want to make sure I get a fresh start with btt before I continue troubleshooting.

I will go get the beta for 10.12.15 too.

**Comment by philosoralphter** [comment]
should i try single user mode first?

**Comment by philosoralphter** [comment]
Ok nothing has worked.  The driver, the 12.14.5 beta, different versions.  The buttons are still registered in places, but not btt.  I'm now at a loss.   

I would suggest that it is something from your most recent push, but i'm pretty sure it actually want' working when I first opened the computer today, and i didn't let btt update till about ten minutes later when I noticed the buttons weren't working.

I guess let me know if you or anyone find anything...

**Comment by philosoralphter** [comment]
alright using the new driver for my mouse did bring back the functionality of the buttons, so they are definitely still working.  Seems like btt doesn't get the click events from those buttons anymore though, or however it works.  I run a very similar setup on my machine at home.  I will see if it's working tonight, and slowly update things till it breaks, and if it does, i'll let you know the details.

**Comment by fifafu** [comment]
You mentioned above that you also tried previous BetterTouchTool versions, thus I think we can eliminate the possibility that it's caused by the latest update.

Unfortunately I'm currently also out of ideas but I'll post here if I find anything.

**Comment by philosoralphter** [comment]
Well I've tried so many things, so not sure how it got fixed, but it did.  
Installed mouse driver (only now just released a driver, relied an built-in drivers before).  That driver manager ui had same option and it worked.  

Strangely, wired version of mouse on home computer, same versions, worked.  So I uninstalled the new driver again, AND upgraded to 2.10.  Then btt recognized the button clicks in the config interface again, and all is well.  Perhaps it was a hardware issue with my machine or some other software temporarily interfering.  

Sorry to waste your time @fifafu, thank you for the responsiveness
